cdap-ui/app/directives/dag-plus/my-dag-factory.js

```diff
@@ -289,7 +289,56 @@ angular.module(PKG.name + '.commons')
       return icon;
     }
 
-    function getGraphLayout(nodes, connections, separation) {
+    function getNodesMap(nodes) {
+      let nodesMap = {};
+      nodes.forEach(node => {
+        nodesMap[node.name] = node;
+      });
+      return nodesMap;
+    }
+
+    function customGraphLayout(graph, nodes, connections) {
+      let graphNodes = graph._nodes;
+      let nodesMap = getNodesMap(nodes);
+
+      angular.forEach(nodes, (node) => {
+        let location = graphNodes[node.name];
+        let locationX = location.x;
+        let locationY = location.y;
+
+        if (node.type === 'alertpublisher' || node.type === 'errortransform') {
+
+          // If the node connecting to this alert publisher/error transform node only has connections
+          // to these types of nodes, then have to push the alert publisher/error transform down a bit more
+          let connToThisNode = connections.find(conn => conn.to === node.name);
+          if (connToThisNode) {
+            let sourceNode = connToThisNode.from;
+            let onlyConnectedToErrorsAlerts = true;
+            for (let i = 0; i < connections.length; i++) {
+              let conn = connections[i];
+              if (conn.from === sourceNode) {
+                let targetNode = nodesMap[conn.to];
+                if (targetNode.type !== 'alertpublisher' && targetNode.type !== 'errortransform') {
+                  onlyConnectedToErrorsAlerts = false;
+                  break;
+                }
+              }
+            }
+            if (onlyConnectedToErrorsAlerts) {
+              locationY += 200;
+            } else {
+              locationY += 70;
+            }
```

Comment:
I think you only need to push them down if there are 3 or less connections from the previous node.

Reply:
Yeah, good find. It's 3 or less connections, if you count connections to error and alert nodes. So I have added the logic to only move down these nodes when there's 0 or 1 non alert or error connections from previous node.